tools/lib/provision.py

```diff
@@ -178,8 +179,30 @@ def main(options):
     os.chdir(ZULIP_PATH)
 
     # setup-apt-repo does an `apt-get update`
-    run(["sudo", "./scripts/lib/setup-apt-repo"])
-    run(["sudo", "apt-get", "-y", "install", "--no-install-recommends"] + APT_DEPENDENCIES[codename])
+    # hash the apt dependencies
+    sha_sum = hashlib.sha1()
+
+    for apt_depedency in APT_DEPENDENCIES[codename]:
+        sha_sum.update(apt_depedency)
+    # hash the content of setup-apt-repo
+    sha_sum.update(subprocess_text_output(['cat', 'scripts/lib/setup-apt-repo']))
+
+    new_apt_dependencies_hash = sha_sum.hexdigest()
+
+    try:
+        hash_file = open('var/apt_dependenices_hash', 'r+')
+    except IOError:
+        run(['touch', 'var/apt_dependenices_hash'])
+        hash_file = open('var/apt_dependenices_hash', 'r+')
+
+    last_apt_dependencies_hash = hash_file.read()
```

Comment:
this can be simplified:

```
try:
    hash_file = open('var/apt_dependenices_hash', 'r+')
    last_apt_dependencies_hash = hash_file.read()
except IOError:
    last_apt_dependencies_hash = None

```